lms/views/dashboard.py

```diff
@@ -44,7 +49,8 @@ def assignment_show(self):
         """
         assignment = self.get_request_assignment()
         self.request.context.js_config.enable_dashboard_mode(assignment)
-        self._set_lti_user_cookie(self.request.response)
+        if self.request.lti_user:
```

Comment:
This conditional should fix the second scenario in https://github.com/hypothesis/lms/pull/6209#discussion_r1576145538